src/frontend/ocamlmerlin/ocamlmerlin.c

```diff
@@ -288,6 +288,10 @@ static void start_server(const char *socketname, const char* eventname, const ch
   ZeroMemory(&si, sizeof(si));
   si.cb = sizeof(si);
   ZeroMemory(&pi, sizeof(pi));
+  /* Change directory to TMPDIR, so that process still works if directory
+     is deleted. */
+  if (GetTempPath(PATHSZ, lpTempDir) == 0 || SetCurrentDirectory(lpTempDir) == 0)
```

Comment:
I lightly suggest `GetTempPath2`.
> When calling this function from a process running as SYSTEM it will return the path C:\Windows\SystemTemp, which is inaccessible to non-SYSTEM processes. For non-SYSTEM processes, GetTempPath2 will behave the same as [GetTempPath](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettemppatha).

Reply:
Thanks, done.